Surefire/Failsafe do not work on path with space(s) after #51796

### Describe the bug

Surefire/Failsafe do not work for newly generated projects on paths with space(s), this is regression introduced by #51796

Build ends with ClassNotFoundException. CC @Sanne @gsmet 

Things work again after adding `<argLine>--add-opens java.base/java.lang=ALL-UNNAMED</argLine>` to the configuration of Surefire/Failsafe.

```
~/Downloads/s p a c e s/app >   mvn -V clean package
...
[INFO] --- surefire:3.5.4:test (default-test) @ app ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
Error: Could not find or load main class p
Caused by: java.lang.ClassNotFoundException: p
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.518 s
[INFO] Finished at: 2026-01-08T09:54:19+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.5.4:test (default-test) on project app:
[ERROR]
[ERROR] See /Users/rsvoboda/Downloads/s p a c e s/app/target/surefire-reports for the individual test results.
[ERROR] See dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd '/Users/rsvoboda/Downloads/s p a c e s/app' && '/Users/rsvoboda/.sdkman/candidates/java/21.0.9-tem/bin/java' '--add-opens=java.base/java.lang.invoke=ALL-UNNAMED' '--add-opens=java.base/java.lang=ALL-UNNAMED' '--add-exports=java.base/jdk.internal.module=ALL-UNNAMED' '-Dquarkus-internal-test.serialized-app-model.path=/Users/rsvoboda/Downloads/s' 'p' 'a' 'c' 'e' 's/app/target/quarkus/bootstrap/test-app-model.dat' '-jar' '/Users/rsvoboda/Downloads/s p a c e s/app/target/surefire/surefirebooter-20260108095419422_3.jar' '/Users/rsvoboda/Downloads/s p a c e s/app/target/surefire' '2026-01-08T09-54-19_390-jvmRun1' 'surefire-20260108095419422_1tmp' 'surefire_0-20260108095419422_2tmp'
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd '/Users/rsvoboda/Downloads/s p a c e s/app' && '/Users/rsvoboda/.sdkman/candidates/java/21.0.9-tem/bin/java' '--add-opens=java.base/java.lang.invoke=ALL-UNNAMED' '--add-opens=java.base/java.lang=ALL-UNNAMED' '--add-exports=java.base/jdk.internal.module=ALL-UNNAMED' '-Dquarkus-internal-test.serialized-app-model.path=/Users/rsvoboda/Downloads/s' 'p' 'a' 'c' 'e' 's/app/target/quarkus/bootstrap/test-app-model.dat' '-jar' '/Users/rsvoboda/Downloads/s p a c e s/app/target/surefire/surefirebooter-20260108095419422_3.jar' '/Users/rsvoboda/Downloads/s p a c e s/app/target/surefire' '2026-01-08T09-54-19_390-jvmRun1' 'surefire-20260108095419422_1tmp' 'surefire_0-20260108095419422_2tmp'
```

Maven / Java details
```
Apache Maven 3.9.12 (848fbb4bf2d427b72bdb2471c22fced7ebd9a7a1)
Maven home: /Users/rsvoboda/.sdkman/candidates/maven/current
Java version: 21.0.9, vendor: Eclipse Adoptium, runtime: /Users/rsvoboda/.sdkman/candidates/java/21.0.9-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "26.2", arch: "aarch64", family: "mac"
```

### Expected behavior

Surefire/Failsafe do not work on path with space(s), this is regression introduced by #51796

### Actual behavior

Surefire/Failsafe work as expected on paths with space(s)

### How to Reproduce?
```
mkdir "s p a c e s"
cd "s p a c e s"
java -jar ~/.m2/repository/io/quarkus/quarkus-cli/999-SNAPSHOT/quarkus-cli-999-SNAPSHOT-runner.jar create app app --platform-bom=io.quarkus::999-SNAPSHOT
cd app
mvn clean package
```

### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

Java 21, Java 25

### Quarkus version or git rev

c5d44d09438

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/GenerateCodeMojo.java
@@ -122,7 +122,8 @@ public class GenerateCodeMojo extends QuarkusBootstrapMojo {
                             Properties properties = mavenProject().getProperties();
                             String argLine = properties.getProperty("argLine", "");
                             properties.setProperty("argLine", argLine +
-                                    " -D" + BootstrapConstants.SERIALIZED_TEST_APP_MODEL + "=" + serializedTestAppModelPath);
+                                    " -D" + BootstrapConstants.SERIALIZED_TEST_APP_MODEL + "=\"" + serializedTestAppModelPath
+                                    + "\"");
                         } catch (IOException e) {
                             getLog().warn("Failed to serialize application model", e);
                         }

